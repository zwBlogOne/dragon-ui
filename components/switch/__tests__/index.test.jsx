import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Switch from '../index';

describe('Switch', () => {
  it('renders normal Switch correctly', () => {
    const wrapper = render(
      <div>
        <Switch />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Switch correctly with more props', () => {
    const wrapper = render(
      <div>
        <Switch disabled size="sm" isCheckedText="是" unCheckedText="否" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when receving new value', () => {
    const wrapper = mount(
      <Switch defaultValue={false} />
    );

    expect(wrapper.find('.ui-switch').hasClass('checked')).toBeFalsy();
    wrapper.setProps({ value: true });
    expect(wrapper.find('.ui-switch').hasClass('checked')).toBeTruthy();
  });

  it('behaves correctly when toggling status', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Switch defaultValue={false} onChange={onChange} />
    );

    wrapper.find('.ui-switch').simulate('click');
    expect(onChange).toHaveBeenCalledWith(true);
  });
});

