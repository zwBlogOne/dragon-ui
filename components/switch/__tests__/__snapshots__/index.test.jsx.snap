// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders Switch correctly with more props 1`] = `
<div>
  <span
    class="ui-switch disabled size-sm"
  >
    <span
      class="ui-switch-inner"
    >
      Âê¶
    </span>
  </span>
</div>
`;

exports[`Switch renders normal Switch correctly 1`] = `
<div>
  <span
    class="ui-switch"
  >
    <span
      class="ui-switch-inner"
    />
  </span>
</div>
`;
